### PR TITLE
Gender component

### PR DESCRIPTION
**Pull request by OxygenO22**

### CODE REVIEW

**Review by Kurilo-Pavel — APPROVED** [review]

File: src/components/GenderSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                            </p>
+                        </div>
+                        <div className={classes.radio__item}> 
+                            <input 
+                                className={classes.radio__inner}
+                                type='radio'
+                                name='Other'
+                                checked={radio === 'Other'}
+                                onChange={e => {
+                                    setRadio(e.target.name);
```
Comment (Evgeniy121212) [inline src/components/GenderSelector.tsx:64]: please, create constants for each selected types and use at the name, checked and here, in setRadio

const GENDER__MALE = 'male'


                            <input 
                                className={classes.radio__inner}
                                type='radio'
                                name={GENDER__MALE}
                                checked={radio === GENDER__MALE}
                                onChange={e => {
                                    setRadio(GENDER__MALE);
                                    setRadioError('');
                                }}
                                {...props} 
                            />

File: src/components/GenderSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                                name='Female'
+                                checked={radio === 'Female'}
+                                onChange={e => {
+                                    setRadio(e.target.name);
+                                    setRadioError('');
+                                }}
+                                {...props} 
+                            />
+                            <p className={classes.radio__text}>
+                                Female
```
Comment (Evgeniy121212) [inline src/components/GenderSelector.tsx:54]: use <label> for input, not <p>

File: src/components/GenderSelector.module.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        .radio__inner {
+          margin-right: 10px;
+        }
+
+        .radio__text {
+          font-size: 1.2em;
+          font-weight: 700;
+        }
```
Comment (Evgeniy121212) [inline src/components/GenderSelector.module.scss:39]: if you use BEM - not include one selector into another - use one level of nesting:

```css
.radio__title {}
.radio__items {}
.radio__item {}
.radio__inner {}
.radio__text {}
```

File: src/components/GenderSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                
+            </label>
+            {/* <input className={classes.button}
+                    type="submit"
+                    value="Finish"
+                    onClick={e => {
+                        e.preventDefault();
+                        blurHandler();
+                    }} 
+                />  */} // for tests
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:85]: done

File: src/components/GenderSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                            <p className={classes.radio__text}>
+                                Other
+                            </p>
+                        </div>
+                    </div>
+                </div>
+                {(radioError) && <div style={{color: 'red'}}>{radioError}</div>}
+                
+            </label>
+            {/* <input className={classes.button}
```
Comment (Terapool) [inline src/components/GenderSelector.tsx:78]: Remove this commented code. Bad approach

File: src/components/GenderSelector.tsx (hunk 1 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:64]: done

File: src/components/GenderSelector.tsx (hunk 2 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:54]: I don't understand, please explain what this means.

File: src/components/GenderSelector.tsx (hunk 5 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:78]: done

**Review by valer-sky — APPROVED** [review]

File: src/components/GenderSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import React, {useState} from 'react';
+import classes from './GenderSelector.module.scss';
+import '../index.scss';
+
+const GenderSelector = (props) => {
```
Comment (Evgeniy121212) [inline src/components/GenderSelector.tsx:5]: add type to the props

**Review by Evgeniy121212 — CHANGES_REQUESTED** [review]

File: src/components/GenderSelector.tsx (hunk 2 above)
Comment (Evgeniy121212) [reply src/components/GenderSelector.tsx:57]: use `<label>` for input, not `<p>`

File: src/components/GenderSelector.module.scss (hunk 3 above)
Comment (Evgeniy121212) [reply src/components/GenderSelector.module.scss]: At least - not use it with the BEM methodology

File: src/components/GenderSelector.tsx (hunk 2 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:57]: done

File: src/components/GenderSelector.module.scss (hunk 3 above)
Comment (OxygenO22) [reply src/components/GenderSelector.module.scss]: done

File: src/components/GenderSelector.tsx (hunk 6 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx]: done

**Review by Evgeniy121212 — APPROVED** [review]

File: src/components/GenderSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+import classes from './GenderSelector.module.scss';
+import '../index.scss';
+
+const GenderSelector: React.FC = (props) => {
+    const [radio, setRadio] = useState<string>('');
+    const [radioError, setRadioError] = useState<string>(''); 
+    const GENDER__MALE = 'male';
+    const GENDER__FEMALE = 'female';
+    const GENDER__OTHER = 'other';
+
```
Comment (Terapool) [inline src/components/GenderSelector.tsx:11]: const GENDER = {
&nbsp;&nbsp;MALE: 'male',
&nbsp;&nbsp;FEMALE: female',
...
}

Comment (OxygenO22) [reply src/components/GenderSelector.tsx:14]: Ok, done

**Pull request merged**